compiler/opt.js

```diff
@@ -4,6 +4,39 @@ import { read_signedLEB128, read_ieee754_binary64 } from './encoding.js';
 import { log } from './log.js';
 import Prefs from './prefs.js';
 
+const hasType = (funcs, pages, type) => {
+  switch (type) {
+    case 'Array':
+      return pages.hasArray;
+    case 'String':
+      return pages.hasString;
+    case 'ByteString':
+      return pages.hasByteString;
+
+    case 'Map':
+    case 'Set':
+    case 'WeakMap':
+    case 'WeakSet':
+    case 'WeakRef':
+    case 'Date':
+    case 'ArrayBuffer':
+    case 'SharedArrayBuffer':
+    case 'Uint8Array':
+    case 'Int8Array':
+    case 'Uint8ClampedArray':
+    case 'Uint16Array':
+    case 'Int16Array':
+    case 'Uint32Array':
+    case 'Int32Array':
+    case 'Float32Array':
+    case 'Float64Array':
+      return funcs.some(x => x.name === type);
```

Comment:
This would break eg `new Uint8Array(8).buffer`?

Reply:
I'm not sure what you mean? That'll still call the constructor, plus this is what the code was doing before, just separated into another function

Reply:
Ignore me, I didn't see that part of the code, I'll remove arraybuffer and sharedarray buffer from the list